node/pkg/lifecycle/startup/startup.go

```diff
@@ -703,6 +703,9 @@ func configureASNumber(node *libapi.Node) bool {
 			return true
 		}
 	} else {
+		if node.Spec.BGP == nil {
+			node.Spec.BGP = &libapi.NodeBGPSpec{}
+		}
 		if node.Spec.BGP.ASNumber == nil {
```

Comment:
I think we probably want `if node.Spec.BGP == nil || node.Spec.BGP.ASNumber == nil` for this branch.

We don't want to initialize the BGP spec to a value if we're not going to set any fields in it - I believe there is validation that prevents that.

Reply:
Avoided initializing the BGP data by returning early as suggested in the other comment.